### ISSUE TITLE
[Screen Reader-Microbit Menu Controls] Control type is defined incorrectly for the Advance Expand/Collapse button. 

### ISSUE THREAD

**Issue by v-mesala** [Microsoft/pxt#2607]
**User Experience:**
Users with Visual disabilities might not able to understand the controls if control type is incorrectly defined. Like in the scenario Control type is defined incorrectly for the Advance Expand/Collapse button. 

**Test Environment**
OS: RS2 Version 1703(OS Build 15063.2)
Platform: Edge.
Screen reader: Narrator

**Repro Steps**
1-Navigate to url:- https://makecode.com/
2-Navigate to the Micro bit section element and select Code control given on it.
3-Navigate to the menu controls listed near the Simulator control.
4-Navigate to the Advance expand/Collapse button and verify the issue.

**Actual Result**
Control type is defined incorrectly for the Advance Expand/Collapse button as Treeitem and State as Aria-Selected = "False" 

**Expected Result**
Control type should be Button and State should be Aria-Expanded = "True" if is expanded and Aria-Expanded = "False" if not expanded. 

**MAS Reference**
https://microsoft.sharepoint.com/teams/msenable/Documents/LCA/MAS%20-%20July%201,%202017%20Release/MAS%204.1.2%20%E2%80%93%20Name,%20Role,%20Value.docx?d=w248054a65e6847719e1e242fb5025730&csf=1

**Suggested fix:**
` 1- Using aria-label to give an invisible name to element, where name/title property is not used 
- e.g.  
- 'Close' button can have only ‘X’, to make it more meaningful aria-label attribute is used with ‘Close’ text to help AT’s without showing the content on UI 
` <div id="box">`
-    This is a pop-up box. 
` <button aria-label="Close" onclick="document.getElementById('box').style.display='none';" class="closebutton"   >`                                    
- 2-   Using ARIA role to expose the control type of user interface component 
 e.g. 
` <div role="toolbar" tabindex="0" id="customToolbar" >`
` <img src="img/btn1.gif" role="button" tabindex="-1" alt="Home" id="b1" title="Home">`
-  
- 3-   Using ARIA attributes to expose the element’s state  
- e.g. 
`<div role="button" id="customToolbar" aria-checked=”” aria-disabled=”” aria-haspopup=”” aria-expanded=”” aria-selected=””>`

**Refer Attachment:**
![mas 4 1 2_makecode_advanced control type](https://user-images.githubusercontent.com/30434417/28820801-a1267d12-76d0-11e7-88ef-6253cb4cd07f.png)

**Comment by veler** [comment]
This is part of Blocky, a third-part component.

**Comment by abchatra** [comment]
We have got exception for Blockly bugs from Julia Liuson (CVP). Closing these as Won't Fix. Following is the reasoning.

----------------------

MakeCode can be used in either Text editor or Block editor.

Blocks editor is inherently visual and is based on google blockly. Usage requires drag and drop functionality and we don’t have a equivalence in MakeCode for this w.r.t accessibility. 
We have evaluated couple of research projects which tries to find a equivalence to block based coding for accessibility needs: 
•	Project Torino (Requires students to buy physical blocks)
•	Accessible blockly (This is text based coding without syntax errors)
Both these are drastic departure from the blockly and provides a complete new experience. These are still in research and we intend not to use them in MakeCode at this point of time.

As workaround, the text editor in MakeCode is the way we see students with the need of accessibility using MakeCode.
